File: client/ios/App/Support/SupportFeatureBridge.swift
import Foundation

public struct SupportFeatureBridge: Sendable {
    private let gateway: SupportGateway

    public init(gateway: SupportGateway) {
        self.gateway = gateway
    }

    public func load(path: String) async throws -> SupportPagePayload {
        try await gateway.load(path: path)
    }

    public func submit(path: String, fields: [String: String]) async throws -> SupportPagePayload {
        try await gateway.submit(path: path, fields: fields)
    }
}

File: client/android/app/src/main/java/app/mobiling/client/support/SupportFeatureBridge.kt
package app.mobiling.client.support

import app.mobiling.client.contract.support.SupportPagePayload
import app.mobiling.client.data.support.SupportGateway

class SupportFeatureBridge(
    private val gateway: SupportGateway,
) {
    suspend fun load(path: String): SupportPagePayload = gateway.load(path)

    suspend fun submit(path: String, fields: Map<String, String>): SupportPagePayload =
        gateway.submit(path, fields)
}
